src/d3a/d3a_core/simulation.py

```diff
@@ -399,19 +399,15 @@ def _execute_simulation(self, slot_resume, tick_resume, console=None):
                 config.sim_duration / (self.progress_info.elapsed_time - paused_duration)
             )
         self._update_and_send_results()
-        if self.export_on_finish and self.should_export_results:
+        if self.export_results_on_finish:
             log.info("Exporting simulation data.")
             self.export.data_to_csv(self.area, False)
             self.export.area_tree_summary_to_json(self.endpoint_buffer.area_result_dict)
-            if GlobalConfig.POWER_FLOW:
-                self.export.export(export_plots=self.should_export_results,
-                                   power_flow=self.power_flow)
-            else:
-                self.export.export(self.should_export_results)
+            self.export.export(power_flow=self.power_flow if GlobalConfig.POWER_FLOW else None)
 
     @property
-    def should_export_results(self):
-        return not self.kafka_connection.is_enabled()
+    def should_send_results(self):
```

Comment:
Nit: the naming/documentation seems to be still a bit unclear to me (where should the results be sent? Why?)

Reply:
I think that should_send_results_to_broker would help clarify this. I would suggest should_publish_result if we sticked to Redis, however we are not running on pubsub anymore, therefore it is a misleading name.